app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java

```diff
@@ -74,11 +86,26 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
                 calendar.get(Calendar.MONTH),
                 calendar.get(Calendar.DAY_OF_MONTH));
 
-        if (isStartOrEnd) {
-            dialog.getDatePicker().setMaxDate(new Date().getTime());
-        } else {
-            dialog.getDatePicker().setMaxDate(new Date().getTime());
-            dialog.getDatePicker().setMinDate(startDate);
+        Bundle args = getArguments();
+        this.datePickerType = args.getInt("datePickerType", this.ALL_DAYS);
+
+        switch (datePickerType) {
+            case ONLY_CURRENT_DAY:
+                dialog.getDatePicker().setMaxDate(new Date().getTime());
+                dialog.getDatePicker().setMinDate(new Date().getTime());
+                break;
+
+            case STARTS_CURRENT_DAY:
+                dialog.getDatePicker().setMinDate(new Date().getTime());
+                break;
+
+            case ENDS_CURRENT_DAY:
+                dialog.getDatePicker().setMaxDate(new Date().getTime());
+                break;
+
+            case ALL_DAYS:
```

Comment:
We don't want this last case, as we are doing nothing here.

Reply:
I created that case in order to prevent bugs... but I'll remove it and will see if there is no problem.